src/infrastructure/implementations/basic-key-manager.ts

```diff
@@ -34,6 +36,19 @@ export class BasicKeyManager implements IKeyManager {
         privateKey = "0".repeat(privateKey.length);
         return selfKey.derive(userKey.getPublic()).toString(16);
     }
+
+    public decryptMessage = async (encryptedMessage: string): Promise<string> => {
```

Comment:
Encryption and Decryption code is now spread around 2 places.

How about we make a new class ECIESEncryption with 2 static methods encrypt & decrypt. We can move the functionality there and refer to that class from BasicKeyManager.decryptMessage & EncryptionManager.encrypt

Reply:
Done